IntegrationExample/Processing/PartnerAudioProcessingController.swift

```diff
@@ -6,137 +6,91 @@
+
     private let session: MimiProcessingSession
     private let firmwareController: FirmwareControlling
     
     private var isEnabledApplicator: MimiProcessingParameterApplicator<Bool>!
     private var intensityApplicator: MimiProcessingParameterApplicator<Float>!
     private var presetApplicator: MimiProcessingParameterApplicator<MimiPersonalization.Preset?>!
-    
+
     private var cancellables = Set<AnyCancellable>()
     
     // MARK: Init
     
-    init(session: MimiProcessingSession, firmwareController: FirmwareControlling) {
+    init(session: MimiProcessingSession, firmwareController: FirmwareControlling) async {
         self.session = session
         self.firmwareController = firmwareController
-        
-        let isEnabledApplicator = makeIsEnabledApplicator(session: session, firmwareController: firmwareController)
-        let intensityApplicator = makeIntensityApplicator(session: session, firmwareController: firmwareController)
-        let presetApplicator = session.preset.applicator()
-        
-        setUpIsEnabledApplicator(isEnabledApplicator)
-        setUpIntensityApplicator(intensityApplicator)
-        setUpPresetApplicator(presetApplicator)
-        
-        // Hold strong references to parameter applicators
-        self.intensityApplicator = intensityApplicator
-        self.isEnabledApplicator = isEnabledApplicator
-        self.presetApplicator = presetApplicator
+
+        await makeIsEnabledApplicator(session: session, firmwareController: firmwareController)
+        await makeIntensityApplicator(session: session, firmwareController: firmwareController)
+        await makePresetApplicator(session: session, firmwareController: firmwareController)
     }
     
-    private func makeIsEnabledApplicator(session: MimiProcessingSession, firmwareController: FirmwareControlling) -> MimiProcessingParameterApplicator<Bool> {
-        let isEnabledValue = firmwareController.getIsEnabled()
-        
+    private func makeIsEnabledApplicator(session: MimiProcessingSession, firmwareController: FirmwareControlling) async {
+        let isEnabledValue = await firmwareController.getIsEnabled()
+
         // Create an isEnabled applicator that is in sync with the isEnabled value on the firmware.
-        let (isEnabledApplicator, _) = session.isEnabled.applicator(synchronizing: isEnabledValue)
-        
-        return isEnabledApplicator
+        self.isEnabledApplicator = await session.isEnabled.addApplicator(apply: { [weak self] value in
+            guard let self else { return }
+
+            // If for any reason e.g. Mimi enabled headphones disconnected, the `value` cannot
+            // be applied then throw an error here.
+
+            try await self.firmwareController.setIsEnabled(value)
+        })
+
+        do {
+            // Sync the applicator with the firmware isEnabled value.
+            try await session.isEnabled.apply(isEnabledValue)
```

Comment:
I have a question: where and when do we need to use the `synchronizeApplicators` function?